### PR TITLE
Archer dev 1

### PR DESCRIPTION
**Pull request by ArcherPTG**
implented docker compose stuff to waddlebot core
also merged wddledmb into the core

## Summary by Sourcery

Integrate WaddleDBM module into the core application and add Docker Compose configuration to support multiple services. Include documentation and licensing for the new module.

New Features:
- Integrate WaddleDBM module into the core application, providing additional database management capabilities.

Build:
- Add Docker Compose configuration to support multiple services including Redis, Matterbridge, and various WaddleBot modules.

Documentation:
- Add a LICENSE file for WaddleDBM module under GNU Affero General Public License.
- Add USAGE.md documentation for WaddleDBM module.

### CODE REVIEW

File: modules/WaddleDBM/entrypoint.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+- name: PROJECT NAME
```
Comment (PenguinzPlays) [inline modules/WaddleDBM/entrypoint.yml:2]:
```suggestion
- name: WaddleDBM
```

File: modules/WaddleDBM/errors/127.0.0.1.2024-06-03.12-24-20.a9aa0fce-3b76-4585-a3c4-20b6da4d70da
Comment (PenguinzPlays) [inline modules/WaddleDBM/errors/127.0.0.1.2024-06-03.12-24-20.a9aa0fce-3b76-4585-a3c4-20b6da4d70da:1]: maybe ignore this directory ? This is spammy and shouldnt be up here.

**Review by PenguinzPlays — CHANGES_REQUESTED** [review]
Lots of changes needed.

File: modules/WaddleBot-Gateway-Manager/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+LABEL org.opencontainers.image.authors="info@penguintech.group"
+LABEL license="GNU AGPL3"
+
+# GET THE FILES WHERE WE NEED THEM!
+COPY . /opt/manager/
+WORKDIR /opt/manager
+
+
+# PUT YER ARGS in here
+ARG APP_TITLE="waddlebot-core"
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Gateway-Manager/Dockerfile:12]: the app title says waddlebot core, im not sure what youre asking, do u want me to change it to gateway manager?

File: modules/WaddleBot-Gateway-Manager/entrypoint.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+- name: PROJECT NAME
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Gateway-Manager/entrypoint.yml]: adjusted

File: jobs/run/run.yml [hunk 4]
```diff
@@ -2,11 +2,16 @@
 - name: Run the WaddleBot-Configurator configure script
   command: python3 /opt/manager/modules/WaddleBot-Configurator/app/configurator.py
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply jobs/run/run.yml:3]: would this work? i replaced the old one with it 

`- name: Run the WaddleBot-Configurator configure script
  ansible.builtin.script: /opt/manager/modules/WaddleBot-Configurator/app/configurator.py
  args:
    executable: python3`

File: modules/WaddleBot-Gateway-Manager/run.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Run the WaddleBot-Configurator configure script
+  command: python3 /opt/manager/app/app.py
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Gateway-Manager/run.yml]: changed to 

`- name: Run the WaddleBot-gateway manager configure script
  ansible.builtin.script: /opt/manager/app/app.py
  args:
    executable: python3`
    
    is it corect?

File: modules/WaddleBot-Twitch-Activity-Listener/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ENV TWITCH_APP_SECRET=""
+ENV CONTEXT_URL="http://127.0.0.1:8000/WaddleDBM/context/initialize_user.json"
+ENV GATEWAYS_GET_URL="http://127.0.0.1:8000/WaddleDBM/routing_gateways/get_all.json"
+ENV EVENTSUB_URL = 'https://mj-dev.penguintech.io/'
+
+
+# Python related commands to install dependencies, create a virtual environment, and run the application
+RUN apt-get update 
+RUN apt-get install -y python3-dev
+RUN apt-get install -y libpq-dev
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Twitch-Activity-Listener/Dockerfile]: so its changed to this in the build file 

- name: Update repositories cache 
  ansible.builtin.apt:
    update_cache: yes

- name: Install python3-dev
  ansible.builtin.apt:
    name: python3-dev
    state: present

- name: Install libpq-dev
  ansible.builtin.apt:
    name: libpq-dev
    state: present

- name: Install postgresql
  ansible.builtin.apt:
    name: postgresql
    state: present

- name: Install specified python requirements
  ansible.builtin.pip:
    requirements: /app/requirements.txt

File: modules/WaddleBot-Twitch-Authenticator/entrypoint.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+---
+- name: PROJECT NAME
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Twitch-Authenticator/entrypoint.yml]: fixed it ill try to fix where i can

File: modules/WaddleBot-Twitch-Activity-Listener/run.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Run the WaddleBot-Configurator configure script
+  command: python3 /opt/manager/app.py
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Twitch-Activity-Listener/run.yml]: changed it

File: modules/WaddleBot-Twitch-Authenticator/run.yml [hunk 9]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Run the WaddleBot-Configurator configure script
+  command: python3 /opt/manager/app.py
```
[the remark this answers is not in the document]
Comment (ArcherPTG) [reply modules/WaddleBot-Twitch-Authenticator/run.yml]: used teh script moduel

**Pull request closed without merging**